doc/hardware/emulator/index.rst

```diff
@@ -1,189 +1,91 @@
-  tests can be performed without access to the real hardware
-
-* SPI emulator driver, which does the same for SPI
-
-* eSPI emulator driver, which does the same for eSPI. The emulator is being
-  developed to support more functionalities.
-
-* CAN loopback driver
-
-A GPIO emulator is planned but is not yet complete.
-
-Samples
-=======
-
-Here are some examples present in Zephyr:
-
-#. Bosch BMI160 sensor driver connected via both I2C and SPI to an emulator:
-
-   .. zephyr-app-commands::
-      :app: tests/drivers/sensor/accel/
-      :board: native_sim
-      :goals: build
-
-#. Simple test of the EEPROM emulator:
-
-   .. zephyr-app-commands::
-      :app: tests/drivers/eeprom/api
-      :board: native_sim
-      :goals: build
-
-#. The same test can be built with a second EEPROM which is an Atmel AT24 EEPROM driver
-   connected via I2C an emulator:
-
-   .. zephyr-app-commands::
-      :app: tests/drivers/eeprom/api
-      :board: native_sim
-      :goals: build
-      :gen-args: -DDTC_OVERLAY_FILE=at2x_emul.overlay -DOVERLAY_CONFIG=at2x_emul.conf
-
-API Reference
-*************
-
-.. doxygengroup:: io_emulators
+**ADC emulator**
+  * A fake driver which pretends to be actual ADC, and can be used for testing higher-level API
+    for ADC devices.
+  * Main Kconfig option: :kconfig:option:`CONFIG_ADC_EMUL`
+  * DT binding: :dtcompatible:`zephyr,adc-emul`
+
+**DMA emulator**
+  * Emulated DMA controller
+  * Main Kconfig option: :kconfig:option:`CONFIG_DMA_EMUL`
+  * DT binding: :dtcompatible:`zephyr,dma-emul`
+
+**EEPROM emulator**
+  * Emulate an EEPROM on a flash partition
+  * Main Kconfig option: :kconfig:option:`CONFIG_EEPROM_EMULATOR`
+  * DT binding: :dtcompatible:`zephyr,emu-eeprom`
+
+**EEPROM simulator**
+  * Emulate an EEPROM on RAM
+  * Main Kconfig option: :kconfig:option:`CONFIG_EEPROM_SIMULATOR`
+  * DT binding: :dtcompatible:`zephyr,sim-eeprom`
+  * Note: For :ref:`native targets <native_sim>` it is also possible to keep the content
+    as a file on the host filesystem.
+
+**External bus and bus connected peripheral emulators**
+  * :ref:`Documentation <bus_emul>`
+  * Allow emulating external buses like I2C or SPI and peripherals connected to them.
+
+**Flash simulator**
+  * Emulate a flash on RAM
+  * Main Kconfig option: :kconfig:option:`CONFIG_FLASH_SIMULATOR`
+  * DT binding: :dtcompatible:`zephyr,sim-flash`
+  * Note: For native targets it is also possible to keep the content as a file on the host
+    filesystem. Check :ref:`the native_sim flash simulator section <nsim_per_flash_simu>`.
+
+**GPIO emulator**
+  * Emulated GPIO controllers which can be driven from SW
+  * Main Kconfig option: :kconfig:option:`CONFIG_GPIO_EMUL`
+  * DT binding: :dtcompatible:`zephyr,gpio-emul`
+
+**I2C emulator**
+  * Emulated I2C bus. See :ref:`bus emulators <bus_emul>`.
+  * Main Kconfig option: :kconfig:option:`CONFIG_I2C_EMUL`
+  * DT binding: :dtcompatible:`zephyr,i2c-emul-controller`
```

Comment:
Nit: I think it's useful to call the specific bus emulator sub-types on this page, but it might be nice to group them immediately below the general "external bus" heading or as explicit sub-headings of that heading. You might also want to include eSPI in this list.

Reply:
Thanks @alevkoy I missed the eSPI one. I just added it in the TODO here: https://github.com/zephyrproject-rtos/zephyr/issues/66530
Would you be so kind as to send a PR adding it?